### PR TITLE
xtensa: Set CONFIG_DCACHE_LINE_SIZE for NXP/Intel boards used in SOF module

### PR DESCRIPTION
**Pull request by LaurentiuM1234**

### CODE REVIEW

**Review by nashif — DISMISSED** [review]

**Review by kv2019i — DISMISSED** [review]
+1 for all Intel bits.

File: boards/xtensa/intel_adsp_ace15_mtpm/intel_adsp_ace15_mtpm_defconfig [hunk 1]
```diff
@@ -12,3 +12,5 @@ CONFIG_GEN_IRQ_VECTOR_TABLE=n
 CONFIG_BUILD_OUTPUT_BIN=n
 CONFIG_MM_DRV=y
 CONFIG_CRYPTO=y
+
+CONFIG_DCACHE_LINE_SIZE=64
```
Comment (jxstelter) [inline boards/xtensa/intel_adsp_ace15_mtpm/intel_adsp_ace15_mtpm_defconfig:16]: There is one more Intel platform intel_adsp_ace20_lnl. Could you add the config to that one also?

Comment (LaurentiuM1234) [reply boards/xtensa/intel_adsp_ace15_mtpm/intel_adsp_ace15_mtpm_defconfig:16]: Sure, I wouldn't mind but this PR is only supposed to target boards used by SOF and as far as I can tell (please do correct me if I'm wrong) that one isn't one of them.

Comment (jxstelter) [reply boards/xtensa/intel_adsp_ace15_mtpm/intel_adsp_ace15_mtpm_defconfig:16]: The PR: https://github.com/thesofproject/sof/pull/7479 is still in review.

Comment (LaurentiuM1234) [reply boards/xtensa/intel_adsp_ace15_mtpm/intel_adsp_ace15_mtpm_defconfig:16]: ACK, will also do the change for that one.

**Comment by softwarecki** [comment]
I don't understand why we need this definition? We already have defined `d-cache-line-size` in dt. It can be read using `sys_cache_data_line_size_get()`

**Comment by LaurentiuM1234** [comment]
> I don't understand why we need this definition? We already have defined `d-cache-line-size` in dt. It can be read using `sys_cache_data_line_size_get()`

Sadly that can't be used in SOF as the size of the data cache line needs to be known at build time as the alignment of some data (e.g: the heap) depends on it.

**Comment by softwarecki** [comment]
> > I don't understand why we need this definition? We already have defined `d-cache-line-size` in dt. It can be read using `sys_cache_data_line_size_get()`
> 
> Sadly that can't be used in SOF as the size of the data cache line needs to be known at build time as the alignment of some data (e.g: the heap) depends on it.

We can still read it from DT at build time. DT_PROP_OR(DT_PATH(cpus, cpu_0), d_cache_line_size, 0);

**Comment by LaurentiuM1234** [comment]
> > > I don't understand why we need this definition? We already have defined `d-cache-line-size` in dt. It can be read using `sys_cache_data_line_size_get()`
> > 
> > 
> > Sadly that can't be used in SOF as the size of the data cache line needs to be known at build time as the alignment of some data (e.g: the heap) depends on it.
> 
> We can still read it from DT at build time. DT_PROP_OR(DT_PATH(cpus, cpu_0), d_cache_line_size, 0);

This will not work for i.MX93 which uses core 1 and introducing more ifdefs in SOF's `zephyr/include/rtos/cache.h` kinda defeats the purpose of this PR.

**Review by dcpleung — APPROVED** [review]

**Review by lyakh — APPROVED** [review]

**Review by iuliana-prodan — APPROVED** [review]

**Pull request merged**